docs/sources/flow/reference/components/discovery.file.md

```diff
@@ -0,0 +1,76 @@
+---
+aliases:
+- /docs/agent/latest/flow/reference/components/discovery.file
+title: discovery.file
+---
+
+# discovery.file
+
+`discovery.file` discovers files on the local filesystem using the [doublestar][] library.
+
+[doublestar]: https://github.com/bmatcuk/doublestar
+
+## Usage
+
+```river
+discovery.file "LABEL" {
+  paths = ["DOUBLESTAR_PATH"]
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+Name | Type           | Description                                                              | Default | Required
+---- |----------------|--------------------------------------------------------------------------|-----| --------
+`paths` | `list(string)` | Doublestar compatible paths to search.                                   |     | yes
+`exclude_paths` | `list(string)` | Doublestar compatible paths to exclude, exclude_paths supercedes paths . |     | no
+`update_period` | `duration`     | How often to sync filesystem and targets.                                | `"10s"` | no
+
+`paths` and `exclude_paths` use [doublestar][] style paths.
+* `/tmp/**/*.log` will match all subfolders of `tmp` and include any files that end in `*.log`.
+* `/tmp/apache/*.log` will match only files in `/tmp/apache/` that end in `*.log`.
+* `/tmp/**` will match all subfolders of `tmp` and `tmp` itself and all files.
+
+
+## Exported fields
+
+The following fields are exported and can be referenced by other components:
+
+Name | Type | Description
+---- | ---- | -----------
+`targets` | `list(map(string))` | The set of targets discovered from the filesystem.
+
+Each target includes the following labels:
+
+* `__path__`: Absolute path to the file.
+
+## Component health
+
+`discovery.file` is only reported as unhealthy when given an invalid
+configuration. In those cases, exported fields retain their last healthy
+values.
+
+## Debug information
+
+`discovery.file` does not expose any component-specific debug information.
+
+### Debug metrics
+
+`discovery.file` does not expose any component-specific debug metrics.
+
+## Examples
+
+This example discovers all files and folders under `/tmp/logs` and the absolute paths are 
+used by `loki.source.file.files` targets.
+
+```river
+discovery.file "files" {
+    paths = ["/tmp/logs/**/*.log"]
+}
+loki.source.file "files" {
+    targets = discovery.file.files.targets
+    forward_to = []
+}
+```
```

Comment:
(Still haven't seen any of the PR code yet, just went straight to the docs)

I think one of the most sought-after usecase would be to tie this with discovery.kubernetes/discovery.relabel to read logs of all pods.

Should we include this in an example so people can easily copy-paste reuse it? Promtail has a similar issue, where it needs a lot of digging to find out the exact magic required for it to work.

Reply:
I thought about that (not specifically k8s but more examples), and maybe it’s fine to put it in reference for now. I know @karengermond has an idea for a section that is task based and that would be perfect for this

Reply:
Ok, at least let's make sure between ourselves that it can fit our purpose (the specific pipeline is a little fuzzy on my mind).

Reply:
As it is now, this isn't enabling use cases with discovery.kubernetes, see my comment here: https://github.com/grafana/agent/pull/2595#discussion_r1043780810